build.rs

```diff
@@ -7,5 +7,11 @@ compile_error!("You have the `http` or `gateway` feature enabled, \
     - `native_tls_backend` uses SChannel on Windows, Secure Transport on macOS, \
     and OpenSSL on other platforms.\n\
     If you are unsure, go with `rustls_backend`.");
+#[cfg(all(feature = "voice", not(any(feature = "secretbox", feature = "sodiumoxide"))))]
+compile_error!("You have the `voice` feature enabled, either the `secretbox` or `sodiumoxide` feature must be selected to let Serenity use `voice`.\n\
+               - `secretbox` uses a pure rust secretbox implementation that will work on any system.\n\
+               - `sodiumoxide` uses an optimized C-library\n\
+               If you are cross-compiling (using --target) or compiling for Windows go with `secretbox`.");
```

Comment:
No need to explain "--target", if someone is cross-compiling, they will know how in the first place.